x/btcstaking/types/create_delegation_parser.go

```diff
@@ -0,0 +1,268 @@
+package types
+
+import (
+	"fmt"
+	math "math"
+
+	"github.com/babylonlabs-io/babylon/btcstaking"
+	bbn "github.com/babylonlabs-io/babylon/types"
+	btcckpttypes "github.com/babylonlabs-io/babylon/x/btccheckpoint/types"
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/wire"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type ParsedPublicKey struct {
+	PublicKey *btcec.PublicKey
+	BbnPk     *bbn.BIP340PubKey
```

Comment:
```suggestion
	*btcec.PublicKey
	*bbn.BIP340PubKey
```
A bit confusing about `BbnPk` because we also call it a BTC public key